File: android-native/FretboardHeatmap/app/src/main/java/com/example/fretboardheatmap/FretLabels.kt
package com.example.fretboardheatmap

/**
 * Labels non-root fretboard notes(for scales) and finger numbers(for chords)
 */
object FretLabels {

    /**
     * Prepare fret labels for heatmap
     * @param topMenu "CHORDS" or "SCALES" choice
     * @param root root note choice
     * @param dropdownChoice dropdown option choice
     * @param string the current string
     * @param fret the current fret
     */
    fun getLabels(
        topMenu: TopMenuChoice?,
        root: String,
        dropdownChoice: String,
        string: Int,
        fret: Int
    ): String {
        return if (topMenu == TopMenuChoice.SCALES) {
            NoteAlphabet.getNoteName(string, fret)
        } else { // CHORDS
            FretPositions.getFingerNumber(dropdownChoice, root, string, fret)
        }
    }
}

File: ios-native/FretboardHeatmap/FretLabels.swift
//
//  FretLabels.swift
//  FretboardHeatmap
//
//  Created by Sean Avery Suguitan on 2026-01-26.
//

import Foundation

///  Labels non-root fretboard notes(for scales) and finger numbers(for chords)
struct FretLabels {
    
    /// Prepare fret labels for heatmap
    /// - Parameters:
    ///   - topMenu: Top menu choice
    ///   - root: Root note choice
    ///   - dropdownChoice: Top menu dropdown choice
    ///   - string: The current string
    ///   - fret: The current fret
    ///  - Returns: Non-root fretboard notes (for scales) and finger numbers(for chords)
    static func getLabels(
        topMenu: menuChoice?,
        root: String,
        dropdownChoice: String,
        string: Int,
        fret: Int
    ) -> String {
        if topMenu == .scales { // musical alphabet
            return NoteAlphabet.getNoteName(string: string, fret: fret)
        } else { // finger positions
            return FretPositions.getFingerNumber(dropdownChoice: dropdownChoice, root: root, string: string, fret: fret)
        }
    }
}
